patroni/dcs/kubernetes.py

```diff
@@ -109,6 +114,8 @@ def server(self):
 
     @property
     def headers(self):
+        if self._token_expires_at and self._token_expires_at <= datetime.datetime.now():
+            self._read_token_file()
```

Comment:
In case something is wrong with the token file the `_read_token_file()` will raise an exception.
Not being properly handled, it will cause Patroni to demote Postgres to a read-only.
With 1m refresh interval and 1h token lifetime, the best strategy would be to continue using the old token.

Reply:
Also, when after updating the token we don't need to call `_make_headers()` but just set the new value of `self._headers['authorization']`.

Reply:
Last, but not least, you forgot to set the new expiration time and hence the token file will be read for every API request after 1m passed with the start of Patroni.

Reply:
Thanks for the quick review! I've pushed a change addressing these points.